packages/gatsby-transformer-remark/src/extend-node-type.js

```diff
@@ -178,8 +178,27 @@ module.exports = (
       } else {
         const ast = await getAST(markdownNode)
         const tocAst = mdastToToc(ast)
+
         let toc
         if (tocAst.map) {
+          const addSlugToUrl = function(node) {
+            if (node.url) {
+              node.url = [
+                store.getState().config.pathPrefix,
```

Comment:
You need to check if the user wants the pathPrefix used e.g. `gatsby build --prefix-paths`. If not, it should build normally without the path prefixing.

Reply:
Oooh I see, I have to use `pathPrefix` directly from the given argument, not go through the store. Thanks!